lib/puppet/util/tcp_validator.rb

```diff
@@ -0,0 +1,36 @@
+require 'socket'
+require 'timeout'
+require 'ipaddr'
+
+module Puppet
+  module Util
+    class TcpValidator
```

Comment:
No special reason. I suspected it was a pattern because both puppetlabs-mongodb[1] and puppetlabs-puppetdb[2] uses it.

[1] https://github.com/puppetlabs/puppetlabs-mongodb/blob/master/lib/puppet/util/mongodb_validator.rb
[2] https://github.com/puppetlabs/puppetlabs-puppetdb/blob/master/lib/puppet/util/puppetdb_validator.rb